sympy/physics/continuum_mechanics/beam.py

```diff
@@ -390,6 +459,129 @@ def applied_loads(self):
         """
         return self._applied_loads
 
+    def _solve_hinge_beams(self, *reactions):
```

Comment:
can you refer the algorithm used i.e. : https://www.asee.org/documents/sections/midwest/2006/Determining_Deflections_of_Hinge-Connected_Beams_Using_Sing.pdf